### PR TITLE
[Backport 2.x] Adding cross version support for registering Transport Actions

### PR DESCRIPTION
**Pull request by saratvemulapalli**
### Description
Manual backport of https://github.com/opensearch-project/OpenSearch/pull/7403

### Related Issues
Part of #7402 

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by VachaShah — APPROVED** [review]

File: CHANGELOG.md
Comment (kotwanikunal) [inline CHANGELOG.md:8]: Can we please wait on it until #7408 gets merged in? 
Or you can drop the CHANGELOG and that will take care of it.

Comment (saratvemulapalli) [reply CHANGELOG.md:8]: Sure. Either way works.

Comment (kotwanikunal) [reply CHANGELOG.md:8]: I think you are better off dropping the file. Saves on the merge conflict :) 
#7408 got merged in with this change.

**Pull request merged**
